ako-infra/ingestion/vcf_k8s_controller.go

```diff
@@ -482,6 +485,32 @@ func (c *VCFK8sController) ValidBootstrapSecretData(controllerIP, secretName, se
 
 	avirest.InfraAviClientInstance(aviClient)
 	utils.AviLog.Infof("Successfully connected to AVI controller using secret provided by NCP")
+
+	// Check if controller version supports VKS Management Service APIs (>= 31.3.1)
+	minVersion, err := utils.NewVersion(avirest.VKSAviVersion)
+	if err != nil {
+		utils.AviLog.Warnf("VKS: Failed to parse minimum required version: %v", err)
+		return true
+	}
+	currentVersion, err := utils.NewVersion(ctrlVersion)
+	if err != nil {
+		utils.AviLog.Warnf("VKS: Failed to parse controller version %s: %v", ctrlVersion, err)
+		return true
+	}
+	if currentVersion.Compare(minVersion) < 0 {
+		utils.AviLog.Infof("VKS: Controller version %s does not support Management Service APIs (requires >= %s)", ctrlVersion, avirest.VKSAviVersion)
+		return true
+	}
+
+	// Create VKS-specific AVI client with version for Management Service APIs using same credentials
+	vksClient, err := avirest.CreateVKSAviClient(controllerIP, username, authToken, caData)
+	if err != nil {
+		utils.AviLog.Warnf("VKS: Failed to create VKS AVI client: %v", err)
+		return false
+	}
+	avirest.VKSAviClientInstance(vksClient)
+	utils.AviLog.Infof("VKS: Successfully initialized VKS AVI client with version %s", avirest.VKSAviVersion)
+
```

Comment:
do we need to check capability or may be env "MANAGED" before doing VKS client init?

Reply:
Yeah, I thought about it @Rishabh-Baranwal. But since it is just client initialization, isn’t it fine? The capability may be enabled at runtime.